builder/virtualbox/vm/step_create_snapshot.go

```diff
@@ -0,0 +1,74 @@
+package vm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	vboxcommon "github.com/hashicorp/packer/builder/virtualbox/common"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type StepCreateSnapshot struct {
+	Name           string
+	TargetSnapshot string
+}
+
+func (s *StepCreateSnapshot) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	driver := state.Get("driver").(vboxcommon.Driver)
+	ui := state.Get("ui").(packer.Ui)
+	if s.TargetSnapshot != "" {
+		time.Sleep(10 * time.Second) // Wait after the Vm has been shutdown, otherwise creating the snapshot might make the VM unstartable
+		ui.Say(fmt.Sprintf("Creating snapshot %s on virtual machine %s", s.TargetSnapshot, s.Name))
+		snapshotTree, err := driver.LoadSnapshots(s.Name)
+		if err != nil {
+			err = fmt.Errorf("Failed to load snapshots for VM %s: %s", s.Name, err)
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		}
+
+		currentSnapshot := snapshotTree.GetCurrentSnapshot()
+		targetSnapshot := currentSnapshot.GetChildWithName(s.TargetSnapshot)
```

Comment:
In several of our other builders, we require a special "force" option in order to delete a snapshot that already exists (see the "force_deregister" option for the amazon builder, the --force flag for deleting output directories that already exist, and issues like https://github.com/hashicorp/packer/issues/7671 where those assumptions were violated).  I'd like to continue doing this so that we don't end up with the situation where we accidentally blow away users' work. Ideally, we'd have a validation step towards the beginning of the build (but not inside Prepare) that makes sure the given target snapshot isn't set if the force flag or force deregister option isn't already set. Not super picky about names, but `force_delete_snapshot` would probably be a good one.

Reply:
I implemented the `force_delete_snapshot` strategy. So the user is required to set this flag to true to overwrite an existing snapshot. The config.go of this builder enforces this.

Reply:
Cool, that's perfect.